File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    dic = {}
    for item, num in zip(want, number):
        dic[item] = num
    
    for i in range(len(discount)-9):
        item_cnt = 0
        tmp_dict = dic.copy()
        for j in range(i,i+10):
            if discount[j] not in tmp_dict.keys():
                continue
            elif 0 < tmp_dict[discount[j]]:
                item_cnt+=1
                tmp_dict[discount[j]]-=1
        
        if item_cnt==10:
            answer+=1
            
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    
    for(let i=0; i<discount.length; i++) {
       
        if(check(want, number, discount.slice(i, i+10))) answer++;
    }
    return answer;
}
function check(want, number, discount) {

    const map = new Map();
    
    for(let i=0; i<discount.length; i++) {
    
        map.has(discount[i]) ? map.set(discount[i], map.get(discount[i])+1) : map.set(discount[i], 1);
    }
    
    for(let i=0; i<want.length; i++) {
        if(isNaN(map.get(want[i]))) return false;
        if(number[i] > map.get(want[i])) return false;
    }
    return true;
}